File: ruby/hamming/hamming.rb
require 'pry'
class Hamming

  def self.compute(strand1, strand2)
    raise_error(strand1, strand2)
    split1 = strand1.split("")
    split2 = strand2.split("")
    counter = 0

    split1.each_with_index do |v, i|
      counter += 1 if v != split2[i]
    end
    
    counter
  end

  def self.raise_error(strand1, strand2)
    raise ArgumentError.new if strand1.length != strand2.length
  end

end

module BookKeeping
  VERSION = 3
end

File: javascript/hamming/hamming.js
function Hamming(){};

Hamming.prototype.compute = function(strand1, strand2){
  var array1 = strand1.split("")
  var array2 = strand2.split("")
  var counter = 0

  function arrayCheck(array1, array2) {
    if (array1.length != array2.length) {
      throw 'DNA strands must be of equal length.';
    }
    return array1.length;
  }

  for (var i = 0; i < arrayCheck(array1, array2); i++) {
    if (array1[i] == array2[i]) {
      continue;
    } else {
      counter += 1;
    }
  }
  return counter;
}

module.exports = Hamming;
